src/browser/BrowserService.cpp

```diff
@@ -803,10 +804,15 @@ QList<Entry*> BrowserService::confirmEntries(QList<Entry*>& pwEntriesToConfirm,
         return {};
     }
 
+    bool bDatabaseChanged = false;
     m_dialogActive = true;
     BrowserAccessControlDialog accessControlDialog;
 
     connect(m_dbTabWidget, SIGNAL(databaseLocked(DatabaseWidget*)), &accessControlDialog, SLOT(reject()));
+    connect(
+        m_dbTabWidget, &DatabaseTabWidget::activateDatabaseChanged, [&bDatabaseChanged]() { bDatabaseChanged = true; });
```

Comment:
I'm surprised this doesn't cause a crash. bDatabaseChanged will be deallocated as soon as this function returns. Also when using lambda functions in connect statements we should always include the "context" which in this case would be `this` right before the lambda. This ensures the lambda function is disconnected if the context is destroyed.

Reply:
Did I understood this correctly: it's enough if I change this to: `connect(
        m_dbTabWidget, &DatabaseTabWidget::activateDatabaseChanged, this, [&bDatabaseChanged]() { bDatabaseChanged = true; });` ?